crypto/ml_kem/ml_kem.c

```diff
@@ -1772,6 +1772,49 @@ int ossl_ml_kem_encode_private_key(uint8_t *out, size_t len,
     return 1;
 }
 
+/*
+ * Stash the seed and or encoded decapsulation key without (yet) performing a
+ * keygen or import, used during decoding, to avoid extra work if we're only
+ * going to export the key again to load into another provider.
+ */
+int ossl_ml_kem_set_prekey(ML_KEM_KEY *key, int flags_set, int flags_clr,
+                           const uint8_t *seed, size_t seed_len,
+                           const uint8_t *dk, size_t dk_len)
+{
+    if (key == NULL
+        || ossl_ml_kem_have_pubkey(key)
+        || ossl_ml_kem_have_seed(key)
+        || key->encoded_dk != NULL
+        || (seed != NULL && seed_len != ML_KEM_SEED_BYTES)
+        || (dk != NULL &&
+            (dk_len != key->vinfo->prvkey_bytes ||
+             (key->encoded_dk = OPENSSL_secure_malloc(dk_len)) == NULL)))
+        return 0;
+
+    if (seed != NULL &&
+        (key->seedbuf = OPENSSL_secure_malloc(seed_len)) == NULL) {
+        if (dk != NULL) {
+            OPENSSL_secure_clear_free(key->encoded_dk, dk_len);
```

Comment:
OPENSSL_secure_free() here - it hasnt done anything other than alloc at this point.